### DISCUSSION TITLE
Change Salesforce field check

### DISCUSSION THREAD

**Discussion by martinherweg** [verbb/formie#1161] [General] [answered]
Hey,

as you may know by now 😅 we are using the Salesforce intergration and had a few problems, one of the more recent ones was that we didn't got field mappings, after looking at the code we realized that the fields need an edit permission on Salesforce so this condition works 
```
if (!$field['updateable']) {
                continue;
            }
```

But for our needs of the integration only `createable` should be needed for my understanding doesn't it?
Or might it be possible to add a setting to see if `updateable` should be allowed?

Also would it be an option when refreshing the integration in the Integrations tab to only call the endpoint for the options that are enabled (Lead, Contact, Opportunity, Account or Case) ?

Thanks and kind Regards,
Martin

**Comment by engram-design** [comment, marked as answer]
So according to the [docs](https://developer.salesforce.com/docs/atlas.en-us.240.0.api.meta/api/sforce_api_calls_describesobjects_describesobjectresult.htm) I would assume this needs to be checked in case the field in question (not just custom fields) is `updateable` and not a read-only field. I'm not too sure about changing that to `createable`, so I'd have to check against that and the various different fields.

As for only refreshing endpoints for the enabled objects, that's something we can look at including.

**Reply by martinherweg** [reply]
Ok, thanks for clarifying!, this was out of a discussion with some salesforce experts who mentioned that to me, but if there are other fields in question I think that is the best way to go.
Maybe it is more reasonable to note it somewhere in the documentation that all fields in question need to be editable.

**Reply by engram-design** [reply]
I'd certainly go with what the Salesforce experts suggest (I'm not one of them!), but at the very least, I'll add it to the docs.
